third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -456,3 +466,36 @@ func resourceDataflowJobIterateMapForceNew(mapKey string, d *schema.ResourceDiff
 		}
 	}
 }
+
+func resourceDataflowJobIterateMapHasChange(mapKey string, d *schema.ResourceData) bool {
+	obj := d.Get(mapKey).(map[string]interface{})
+	for k := range obj {
+		entrySchemaKey := mapKey + "." + k
+		if d.HasChange(entrySchemaKey) {
+			return true
+		}
+	}
+	return false
+}
+
+func resourceDataflowJobIsVirtualUpdate(d *schema.ResourceData) bool {
+	// on_delete is the only virtual field
+	if d.HasChange("on_delete") {
+		// Check if other fields have changes, which would require an actual update request
+		resourceSchema := resourceDataflowJob().Schema
+		for field, fieldSchema := range resourceSchema {
+			if field == "on_delete" {
+				continue
+			}
+			// Each key within a map must be checked for a change
```

Comment:
Is this true for sure? We have other resources with maps that we just do a HasChange on the whole thing for.

Reply:
After some digging, turns out that this is just needed for the `labels` field due to the resourceDataflowJobLabelDiffSuppress. When we decide to suppress a key, the parent field still reports that it has a change. Terraform plan will show no changes, but `d.HasChange("labels") = true`. I ran into this disparity between the `labels` parent/keys and incorrectly assumed that it could happen to other maps.

 I can't find ways to clear the change during the diff suppress, so I believe the iteration is still needed for `labels`. I modified the condition and comments to be specific for the `labels` field.